### PR TITLE
[RLlib] Make Tune trial ID available in EnvRunners (and callbacks).

### PR DESCRIPTION
**Pull request by sven1977**
Make Tune trial ID available in EnvRunners (and callbacks).
* EnvRunners get a `self.tune_trial_id` property.
* This property is available in all env-runner based callbacks via the `env_runner` arg (access `env_runner.tune_trial_id` inside your callbacks).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/env/single_agent_env_runner.py [hunk 1]
```diff
@@ -63,7 +63,8 @@ def __init__(self, config: AlgorithmConfig, **kwargs):
         """
         super().__init__(config=config)
 
-        self.worker_index = kwargs.get("worker_index")
+        self.worker_index: int = kwargs.get("worker_index")
+        self.tune_trial_id: str = kwargs.get("tune_trial_id")
```
Comment (simonsays1980) [inline rllib/env/single_agent_env_runner.py:67]: For what do we need the trial ID in the env runners? For callbacks?

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: rllib/env/single_agent_env_runner.py (hunk 1 above)
Comment (sven1977) [reply rllib/env/single_agent_env_runner.py:67]: Some users might need this in their env.

**Comment by grizzlybearg** [comment]
Thanks for this. This will be very useful in keeping track of details of episodes within trials

**Pull request merged**